core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -186,6 +188,67 @@ private[spark] object JettyUtils extends Logging {
     contextHandler
   }
 
+  /** Create a handler for proxying request to Workers and Application Drivers */
+  def createProxyHandler(
```

Comment:
+1 to targeted tests of string manipulation code.

Move the string manipulation part into its own method, and write targeted unit tests for it.